pkg/api/handler/helpers.go

```diff
@@ -124,16 +124,13 @@ func convertMerkleRootsRequest(beefMerkleRoots []beef.MerkleRootVerificationRequ
 	return merkleRoots
 }
 
-func findStatusByTxID(txID string, statuses []*metamorph.TransactionStatus) *metamorph.TransactionStatus {
-	for _, status := range statuses {
-		if status.TxID == txID {
-			return status
+func filterStatusesByTxIDs(txIDs []string, allStatuses []*metamorph.TransactionStatus) []*metamorph.TransactionStatus {
+	if len(txIDs) == 1 && len(allStatuses) == 1 {
+		if allStatuses[0].TxID == txIDs[0] {
+			return allStatuses
 		}
 	}
```

Comment:
Why? I would remove this check.

Reply:
I believe that the endpoint for submitting individual transactions is used as frequently as the one for submitting entire batches. Given that the system is designed to process a large number of requests per second, this optimization makes sense and offers noticeablebenefits.

Benchmark results:
1. Optimized version
```
pkg: github.com/bitcoin-sv/arc/pkg/api/handler
cpu: AMD Ryzen 7 5800H with Radeon Graphics         
BenchmarkFilterStatusesByTxIDs/txIDs-1-16         	350659722	         3.443 ns/op	       0 B/op	       0 allocs/op
BenchmarkFilterStatusesByTxIDs/txIDs-10-16        	 1630885	       739.2 ns/op	     539 B/op	       6 allocs/op
BenchmarkFilterStatusesByTxIDs/txIDs-100-16       	  100819	     12566 ns/op	   12477 B/op	      16 allocs/op
BenchmarkFilterStatusesByTxIDs/txIDs-1000-16      	   10000	    100461 ns/op	  108585 B/op	      32 allocs/op
BenchmarkFilterStatusesByTxIDs/txIDs-10000-16     	    1140	   1006422 ns/op	 1017428 B/op	     119 allocs/op
PASS
ok  	github.com/bitcoin-sv/arc/pkg/api/handler	7.275s
```
2. Previous version
```
pkg: github.com/bitcoin-sv/arc/pkg/api/handler
cpu: AMD Ryzen 7 5800H with Radeon Graphics         
BenchmarkFilterStatusesByTxIDs/txIDs-1-16         	29270505	        40.69 ns/op	       8 B/op	       1 allocs/op
BenchmarkFilterStatusesByTxIDs/txIDs-10-16        	 1663438	       727.7 ns/op	     539 B/op	       6 allocs/op
BenchmarkFilterStatusesByTxIDs/txIDs-100-16       	  108610	     11171 ns/op	   12476 B/op	      16 allocs/op
BenchmarkFilterStatusesByTxIDs/txIDs-1000-16      	   10000	    100694 ns/op	  108577 B/op	      32 allocs/op
BenchmarkFilterStatusesByTxIDs/txIDs-10000-16     	    1195	   1023224 ns/op	 1017330 B/op	     119 allocs/op
PASS
ok  	github.com/bitcoin-sv/arc/pkg/api/handler	6.947s
```
It is clear that for individual transactions, we save a considerable amount of time, which can be significant at scale. Moreover, the optimization does not significantly affect code readability or performance for arrays longer than one.

Reply:
You're right, 3.44ns compared to 40.69ns is a significant difference if we're expecting higher loads on single transactions submitting. Thanks a lot for doing the benchmarks.